semanticdb/javac/src/main/scala/scala/meta/internal/semanticdb/javac/semantics/Elements.scala

```diff
@@ -0,0 +1,243 @@
+          val params = elem.paramElements.map(_.sym)
+          val tparams = elem.typeParamElements.map(_.sym)
+          s.MethodSignature(
+            typeParameters = Some(s.Scope(symlinks = tparams)),
+            parameterLists = Seq(s.Scope(symlinks = params)),
+            returnType = returnType
+          )
+        case elem: TypeElement =>
+          val parents = {
+            val superclass = elem.getSuperclass
+            val extendParent =
+              if (superclass.getKind == TypeKind.NONE) List(ObjectType)
+              else List(superclass.tpe)
+            val implementationParents = elem.getInterfaces.asScala.map(_.tpe)
+            extendParent ++ implementationParents
+          }
+          val decls = {
+            val (synthetics, others) = elem.enclosedElements.partition(_.isSynthetic)
+            (synthetics ++ others).map(_.sym)
+          }
+          val tparams = elem.typeParamElements.map(_.sym)
+          s.ClassSignature(
+            typeParameters = Some(s.Scope(symlinks = tparams)),
+            parents = parents,
+            declarations = Some(s.Scope(symlinks = decls))
+          )
+        case elem: TypeParameterElement =>
+          val bounds = {
+            val elemBounds =
+              elem.getBounds.asScala.map(_.tpe)
+            elemBounds match {
+              case Seq() => ObjectType
+              case Seq(b) => b
+              case elemBounds => s.IntersectionType(elemBounds)
+            }
+          }
+          s.TypeSignature(
+            upperBound = bounds
+          )
+        case elem: VariableElement if elem.getKind == ElementKind.PARAMETER =>
+          val parent = elem.getEnclosingElement.asInstanceOf[ExecutableElement]
+          val tpe = {
+            val parentParams =
+              parent.getParameters.asInstanceOf[java.util.List[VariableElement]].asScala
+            if (parent.isVarArgs && elem == parentParams.last) {
+              val containedType = elem.asType().tpe match {
+                case s.TypeRef(_, _, Seq(contained)) => contained
+              }
+              s.RepeatedType(containedType)
+            } else elem.asType().tpe
+          }
+          s.ValueSignature(
+            tpe = tpe
+          )
+        case elem: VariableElement =>
+          s.ValueSignature(
+            tpe = elem.asType().tpe
+          )
+        case _ => s.NoSignature
+      }
+    }
+
+    def info: s.SymbolInformation = s.SymbolInformation(
+      symbol = sym,
+      language = s.Language.JAVA,
+      kind = kind,
+      name = name,
+      annotations = annotations,
+      accessibility = accessibility,
+      properties = properties,
+      signature = signature
+    )
+
+    def populateInfos(infos: mutable.ListBuffer[s.SymbolInformation]): s.SymbolInformation = {
+      val myInfo = info
+      infos += myInfo
+      elem match {
+        case elem: TypeElement =>
+          elem.enclosedElements.foreach { elem =>
```

Comment:
What happens to static initializers here? I think metacp excludes them but there are no tests for it. Static initializers have `<cinit>` name and no signature.

Reply:
Javac also excludes them (they are not listed in "enclosed elements'), I can try to figure out how to avoid that but it doesn't look trivial.